cmd/internal/su.go

```diff
@@ -47,3 +41,64 @@ func AutoSu() {
 	}
 	os.Exit(stat.ExitCode())
 }
+
+func trySudo() (path string, arg []string) {
+	pathSudo, err := exec.LookPath("sudo")
+	if err != nil {
+		// fallback
+		var possibleSudoPath = []string{
+			"/usr/bin/sudo", "/usr/sbin/sudo",
+		}
+		var found = false
+		for _, v := range possibleSudoPath {
+			if isExistAndExecutable(v) {
+				pathSudo = v
+				found = true
+				break
+			}
+		}
+		if !found {
+			return "", nil
+		}
+	}
+	// https://github.com/WireGuard/wireguard-tools/blob/71799a8f6d1450b63071a21cad6ed434b348d3d5/src/wg-quick/linux.bash#L85
+	return pathSudo, []string{
+		pathSudo,
+		"-E",
+		"-p",
+		fmt.Sprintf("%v must be run as root. Please enter the password for %%u to continue: ", filepath.Base(os.Args[0])),
+		"--",
+	}
+}
+
+func tryDesktopSudo() (path string, arg []string) {
+	// https://specifications.freedesktop.org/desktop-entry-spec/latest
+	desktop := os.Getenv("XDG_CURRENT_DESKTOP")
```

Comment:
You can check the existence of `pkexec` directly instead of checking `XDG_CURRENT_DESKTOP` first.

Reply:
While pkexec can be used without a desktop environment, checking XDG_CURRENT_DESKTOP first helps identify the preferred authentication method based on the user's environment. However, I'm open to removing this check if there are important use cases for pkexec in non-desktop scenarios. Could you share examples where this would be beneficial?